src/Agent.Worker/Build/ArtifactCommandExtension.cs

```diff
@@ -180,11 +187,21 @@ private void ProcessArtifactUploadCommand(IExecutionContext context, Dictionary<
                 return;
             }
 
+            string uploadService;
+            if (!eventProperties.TryGetValue(ArtifactUploadEventProperties.UploadService, out uploadService) ||
+                string.IsNullOrEmpty(uploadService))
+            {
+                uploadService = ArtifactUploadSerivce.Container;
+            }
+
             // queue async command task to associate artifact.
             context.Debug($"Upload artifact: {fullPath} to server for build: {buildId.Value} at backend.");
             var commandContext = HostContext.CreateService<IAsyncCommandContext>();
             commandContext.InitializeCommandContext(context, StringUtil.Loc("UploadArtifact"));
-            commandContext.Task = UploadArtifactAsync(commandContext,
+
+            if (uploadService == ArtifactUploadSerivce.Container)
```

Comment:
Looks like `string.Equals(..., StringComparison.OrdinalIgnoreCase)` should be used for most string comparisons in the agent.